src/patternfly/components/Tabs/tabs.scss

```diff
@@ -112,167 +126,305 @@
-    }
-  }
-
-  // Secondary tab states
-  .pf-m-tabs-secondary & .pf-c-tabs__button {
-    // Remove current state and hover state
-    &::before,
-    &::after,
-    &:hover::after {
-      content: none;
-    }
-  }
-
-  // Secondary tab hover state
-  .pf-m-tabs-secondary &:not(.pf-m-current) .pf-c-tabs__button:hover {
-    color: var(--pf-c-tabs__item--hover--Color);
+    border-style: var(--pf-c-tabs__button--before--BorderStyle);
+    border-width: 0;
   }
 }
 
-
 // Tab button
 .pf-c-tabs__button {
   position: relative;
-  display: inline-block;
+  display: flex;
+  flex: 1;
   padding: var(--pf-c-tabs__button--PaddingTop) var(--pf-c-tabs__button--PaddingRight) var(--pf-c-tabs__button--PaddingBottom) var(--pf-c-tabs__button--PaddingLeft);
-  font-weight: var(--pf-c-tabs__button--FontWeight);
   color: var(--pf-c-tabs__button--Color);
   user-select: none;
-  background: var(--pf-c-tabs__button--Background);
+  background-color: var(--pf-c-tabs__button--Background);
   border: 0;
   outline-offset: var(--pf-c-tabs__button--OutlineOffset);
 
-  // Tab button borders
   &::before {
-    position: absolute;
-    top: 0;
-    right: 0;
-    bottom: 0;
-    left: 0;
-    content: "";
-    border: solid var(--pf-c-tabs__item--BorderColor);
-    border-width: var(--pf-c-tabs__item--BorderWidth) 0 var(--pf-c-tabs__item--BorderWidth) var(--pf-c-tabs__item--BorderWidth);
+    border-color: var(--pf-c-tabs__button--before--BorderColor);
+  }
+
+  &::after {
+    border-bottom-color: var(--pf-c-tabs__button--after--BorderColor);
+    border-bottom-width: var(--pf-c-tabs__button--after--BorderWidth);
+  }
+
+  // Tab item hover state
+  &:hover::after {
+    --pf-c-tabs__button--after--BorderColor: var(--pf-c-tabs__button--hover--after--BorderColor);
   }
 
   &.pf-m-hover,
   &:hover {
     text-decoration: none;
   }
-}
 
-.pf-c-tabs__scroll-item {
-  position: absolute;
-  top: 0;
-  bottom: 0;
-  width: var(--pf-c-tabs__scroll-button--Width);
+  .pf-c-tabs__item-icon,
+  .pf-c-tabs__item-text {
+    margin-right: var(--pf-c-tabs__button--child--MarginRight);
 
-  &:first-of-type {
-    left: 0;
+    &:last-child {
+      --pf-c-tabs__button--child--MarginRight: 0;
+    }
   }
 
-  &:last-of-type {
-    right: 0;
+  .pf-c-tabs.pf-m-roomy & {
+    --pf-c-tabs__button--PaddingRight: var(--pf-c-tabs--m-roomy--c-tabs__button--PaddingRight);
+    --pf-c-tabs__button--PaddingLeft: var(--pf-c-tabs--m-roomy--c-tabs__button--PaddingLeft);
   }
 }
 
 // Scroll buttons
 .pf-c-tabs__scroll-button {
-  position: absolute;
-  top: 0;
-  bottom: 0;
-  z-index: var(--pf-c-tabs__scroll-button--ZIndex);
+  flex: none;
   width: var(--pf-c-tabs__scroll-button--Width);
   line-height: 1;
-  background-color: var(--pf-c-tabs__item--BackgroundColor);
-  border: 0;
-
-  // Align right scroll button
-  &:nth-of-type(2) {
-    right: 0;
+  color: var(--pf-c-tabs__scroll-button--Color);
+  visibility: hidden;
```

Comment:
is `visibility: hidden;` enough to hide them from all screen reader devices? I thought we needed to use a combination of `display: none; visibility: hidden;`